core/regions/src/main/java/software/amazon/awssdk/regions/ServiceEndpointKey.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.regions;
+
+import java.util.Collection;
+import java.util.Set;
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.Mutable;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.regions.internal.DefaultServiceEndpointKey;
+
+/**
+ * A ServiceEndpointKey uniquely identifies a service endpoint, and can be used to look up endpoints via
+ * {@link ServiceMetadata#endpointFor(ServiceEndpointKey)}.
+ *
+ * <p>An endpoint is uniquely identified by the {@link Region} of that service and the {@link EndpointTag}s associated with that
+ * endpoint. For example, the {@link EndpointTag#FIPS} endpoint in {@link Region#US_WEST_2}.
+ */
+@SdkPublicApi
+@Immutable
+public interface ServiceEndpointKey {
```

Comment:
Do we need an interface since this looks like a data class? Same comment for PartitionEndpointKey

Reply:
Done